### PR TITLE
tests: new websocket server

### PR DESCRIPTION
**Pull request by nicolasburtey**

### CODE REVIEW

**Comment by nicolasburtey** [comment]
api.blink.sv/graphql -> main graphql
ws.blink.sv/graphql -> websocket server

File: src/servers/ws-server.ts [hunk 1]
```diff
@@ -45,26 +45,28 @@ const getContext = async (ctx: Context) => {
   const ipString = isProd
     ? connectionParams?.["x-real-ip"] || connectionParams?.["x-forwarded-for"]
     : connectionParams?.ip
+    ? connectionParams?.ip
+    : ctx.extra
+    ? (ctx.extra as Extra).request?.socket?.remoteAddress
```
Comment (nicolasburtey) [inline src/servers/ws-server.ts:50]: casting is generally a bad practice. why is that needed?

File: src/servers/ws-server.ts [hunk 2]
```diff
@@ -45,26 +45,28 @@ const getContext = async (ctx: Context) => {
...
-  //   const tokenPayload = {
-  //     sub: kratosCookieRes.kratosUserId,
-  //   }
-  //   return sessionContext({
-  //     tokenPayload,
-  //     ip: request?.socket?.remoteAddress,
-  //     body: null,
-  //   })
-  // }
+  const cookies = ctx.extra ? (ctx.extra as Extra).request.headers.cookie : undefined
```
Comment (nicolasburtey) [inline src/servers/ws-server.ts:57]: casting is generally a bad practice. why is that needed?

File: src/servers/ws-server.ts (hunk 1 above)
Comment (ntheile) [reply src/servers/ws-server.ts]: yep, I was trying to get the e2e tests to pass and it kept failing with an IP address parsing error, so I committed this to see if it would pass. I cleaned up the code with a fully typed context and nullish coalescing operators.

**Review by nicolasburtey — COMMENTED** [review]
approved (but can't approved from github because I opened the PR)

**Review by ntheile — DISMISSED** [review]

**Review by ntheile — APPROVED** [review]
